### PR TITLE
ci: update e2e pipeline

### PR DESCRIPTION
**Pull request by greged93**
Time spent on this PR: 1 day

Resolves: #NA

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [x] Testing

# What is the new behavior?
Updates the end 2 end script, making it work with the latest version of Kakarot.

# Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Review by tcoratger — DISMISSED** [review]

File: .github/workflows/e2e-pipeline.yml [hunk 1]
```diff
@@ -34,19 +50,19 @@ jobs:
...
-          git diff --no-patch --exit-code -- deployments/kakarot-staging/
+          git diff --no-patch --exit-code -- deployments/
           has_diff=$?
           set -e
           echo "has_diff=$has_diff" >> $GITHUB_ENV
       - name: Create PR for deployment updates
         if: ${{ env.has_diff == 1 }}
         env:
-          GH_TOKEN: ${{ secrets.GH_TOKEN }}
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (ClementWalter) [inline .github/workflows/e2e-pipeline.yml:65]: is the default GITHUB_TOKEN allowed to PR on kakaort ? I don't think so, but we know have a kakarot bot that you can use, see also the caro update workflows

Comment (Eikix) [reply .github/workflows/e2e-pipeline.yml:65]: what is the status of this

Comment (greged93) [reply .github/workflows/e2e-pipeline.yml:65]: working on it now that I have some time to focus on the CI and the release

**Comment by Eikix** [comment]
trunk fails

File: .github/workflows/e2e-pipeline.yml [hunk 2]
```diff
@@ -1,7 +1,23 @@
...
+        type: choice
+        options:
+          - sepolia
+          - staging
+      action:
+        description: Action to perform on the selected environment
+        required: true
+        type: choice
+        options:
+          - deploy
```
Comment (Eikix) [inline .github/workflows/e2e-pipeline.yml:19]: what does deploy do?

File: scripts/e2e_pipeline.sh [hunk 3]
```diff
@@ -50,59 +50,69 @@ if [ -z "${ENV}" ]; then
...
+		echo "Please provide the STARKNET_SEPOLIA_STAGING_ACCOUNT_ADDRESS environment variable."
+		exit 1
+	fi
+	if [ -z "${STARKNET_SEPOLIA_STAGING_PRIVATE_KEY}" ]; then
+		echo "Please provide the STARKNET_SEPOLIA_STAGING_PRIVATE_KEY environment variable."
 		exit 1
 	fi
 
-	SKIP="--ignore tests/end_to_end/L1L2Messaging"
+	SKIP="--ignore tests/end_to_end/L1L2Messaging --ignore tests/end_to_end/CairoPrecompiles -k 'not test_should_set_account_bytecode and not test_should_set_account_nonce and not test_should_upgrade_account_class and not test_should_raise_when_class_hash_is_not_declared and not test_should_upgrade_class_hash and not test_should_transfer_ownership and not test_should_return_transaction_count and not test_should_withdraw_all_eth'"
```
Comment (Eikix) [inline scripts/e2e_pipeline.sh:98]: why are we skipping?

File: .github/workflows/e2e-pipeline.yml (hunk 2 above)
Comment (greged93) [reply .github/workflows/e2e-pipeline.yml:19]: deploys/upgrades the kakarot contracts

File: scripts/e2e_pipeline.sh (hunk 3 above)
Comment (greged93) [reply scripts/e2e_pipeline.sh:98]: used to be ran without the multisig so skipped them. do you want to keep them knowing that failure of some test might transfer owner ship or upgrade contracts?

Comment (Eikix) [reply scripts/e2e_pipeline.sh:98]: so now that they work with multisig we can keep them?

you mean there is a case where the contract might have ownership transferred and rekt?

**Review by ClementWalter — APPROVED** [review]

**Pull request merged**